cmd/gitops/create/dashboard/cmd.go

```diff
@@ -0,0 +1,255 @@
+		}
+
+		log := clilogger.NewCLILogger(output)
+
+		log.Generatef("Generating GitOps Dashboard manifests ...")
+
+		var secret string
+
+		if flags.Password != "" {
+			secret, err = run.GenerateSecret(log, flags.Password)
+			if err != nil {
+				return err
+			}
+		}
+
+		dashboardName := args[0]
+
+		adminUsername := flags.Username
+
+		if adminUsername == "" && flags.Password != "" {
+			adminUsername = defaultAdminUsername
+		}
+
+		manifests, err := run.CreateDashboardObjects(log, dashboardName, flags.Namespace, adminUsername, secret, "")
+		if err != nil {
+			return fmt.Errorf("error creating dashboard objects: %w", err)
+		} else {
+			log.Successf("Generated GitOps Dashboard manifests")
+			fmt.Println("---")
+			fmt.Println(resourceToString(manifests))
+		}
+
+		if flags.Export {
+			return nil
+		}
+
+		if flags.KubeConfig != "" {
+			kubeConfigArgs.KubeConfig = &flags.KubeConfig
+
+			if flags.Context == "" {
+				log.Failuref("A context should be provided if a kubeconfig is provided")
+				return cmderrors.ErrNoContextForKubeConfig
+			}
+		}
+
+		log.Actionf("Checking for a cluster in the kube config ...")
+
+		var contextName string
+
+		if flags.Context != "" {
+			contextName = flags.Context
+		} else {
+			_, contextName, err = kube.RestConfig()
+			if err != nil {
+				log.Failuref("Error getting a restconfig: %v", err.Error())
+				return cmderrors.ErrNoCluster
+			}
+		}
+
+		cfg, err := kubeConfigArgs.ToRESTConfig()
+		if err != nil {
+			return fmt.Errorf("error getting a restconfig from kube config args: %w", err)
+		}
+
+		kubeClientOpts := run.GetKubeClientOptions()
+		kubeClientOpts.BindFlags(cmd.Flags())
+
+		kubeClient, err := run.GetKubeClient(log, contextName, cfg, kubeClientOpts)
+		if err != nil {
+			return cmderrors.ErrGetKubeClient
+		}
+
+		log.Actionf("Checking if Flux is already installed ...")
+
+		ctx, cancel := context.WithTimeout(context.Background(), flags.Timeout)
+		defer cancel()
+
+		if fluxVersion, err := run.GetFluxVersion(log, ctx, kubeClient); err != nil {
+			log.Failuref("Flux is not found")
+			return err
+		} else {
+			log.Successf("Flux version %s is found", fluxVersion)
+		}
+
+		log.Actionf("Applying GitOps Dashboard manifests")
+
+		man, err := run.NewManager(log, ctx, kubeClient, kubeConfigArgs)
+		if err != nil {
+			log.Failuref("Error creating resource manager")
+			return err
+		}
+
+		err = run.InstallDashboard(log, ctx, man, manifests)
+		if err != nil {
+			return fmt.Errorf("gitops dashboard installation failed: %w", err)
+		} else {
+			log.Successf("GitOps Dashboard has been installed")
+		}
+
+		log.Actionf("Request reconciliation of dashboard (timeout %v) ...", flags.Timeout)
+
+		log.Waitingf("Waiting for GitOps Dashboard reconciliation")
+
+		dashboardPort := "9001"
+		dashboardPodName := dashboardName + "-" + helmChartName
+
+		if err := run.ReconcileDashboard(kubeClient, dashboardName, flags.Namespace, dashboardPodName, flags.Timeout, dashboardPort); err != nil {
```

Comment:
The are used when getting the dashboard pod with the `GetPodFromSpecMap` function. Which was originally added by @chanwit for enabling port forwarding and I reused it. What would be the best way to get the pod instead of using this function?

Reply:
OK, thanks, will try it.

Reply:
Renamed `GetPodFromSpecMap ` function to  `GetPodFromResourceDescription` and made it accept only namespaced name and kind without the whole specmap with the port number.